### PR TITLE
Add parsepath logic to password in various places

### PR DESCRIPTION
**Pull request by jefferai**
* Account password create/set/change
* Authenticate

Also fixes a bug where confirmation on a change-password call would run
for the current password instead of the new one.

### CODE REVIEW

File: internal/cmd/commands/accountscmd/funcs.go [hunk 1]
```diff
@@ -116,59 +117,92 @@ func extraFlagsFuncImpl(c *Command, _ *base.FlagSets, f *base.FlagSet) {
...
+			if strings.TrimSpace(value) != strings.TrimSpace(confirmation) {
+				c.UI.Error("Entered password and confirmation value did not match.")
+				return false
+			}
+			c.flagPassword = strings.TrimSpace(value)
+
+		default:
+			password, err := parseutil.ParsePath(c.flagPassword)
+			if err != nil && err.Error() != parseutil.ErrNotAUrl.Error() {
+				c.UI.Error("Error parsing password flag: " + err.Error())
```
Comment (irenarindos) [inline internal/cmd/commands/accountscmd/funcs.go:146]:
```suggestion
				c.UI.Error(fmt.Sprintf("Error parsing password flag: %s", err.Error()))
```

File: internal/cmd/commands/accountscmd/funcs.go [hunk 2]
```diff
@@ -116,59 +117,92 @@ func extraFlagsFuncImpl(c *Command, _ *base.FlagSets, f *base.FlagSet) {
...
+				c.UI.Error(fmt.Sprintf("An error occurred attempting to read the password. The raw error message is shown below but usually this is because you attempted to pipe a value into the command or you are executing outside of a terminal (TTY). The raw error was:\n\n%s", err.Error()))
+				return false
+			}
+			fmt.Print("Please enter it one more time for confirmation: ")
+			c.flagCurrentPassword = strings.TrimSpace(value)
+
+		default:
+			password, err := parseutil.ParsePath(c.flagCurrentPassword)
+			if err != nil && err.Error() != parseutil.ErrNotAUrl.Error() {
+				c.UI.Error("Error parsing current password flag: " + err.Error())
```
Comment (irenarindos) [inline internal/cmd/commands/accountscmd/funcs.go:169]:
```suggestion
				c.UI.Error(fmt.Sprintf("Error parsing current password flag: %s", err.Error()))
```

File: internal/cmd/commands/accountscmd/funcs.go [hunk 3]
```diff
@@ -116,59 +117,92 @@ func extraFlagsFuncImpl(c *Command, _ *base.FlagSets, f *base.FlagSet) {
...
+			if strings.TrimSpace(value) != strings.TrimSpace(confirmation) {
+				c.UI.Error("Entered password and confirmation value did not match.")
+				return false
+			}
+			c.flagNewPassword = strings.TrimSpace(value)
+
+		default:
+			password, err := parseutil.ParsePath(c.flagNewPassword)
+			if err != nil && err.Error() != parseutil.ErrNotAUrl.Error() {
+				c.UI.Error("Error parsing new password flag: " + err.Error())
```
Comment (irenarindos) [inline internal/cmd/commands/accountscmd/funcs.go:201]:
```suggestion
				c.UI.Error(fmt.Sprintf("Error parsing new password flag: %s", err.Error()))
```

File: internal/cmd/commands/accountscmd/password_funcs.go [hunk 4]
```diff
@@ -114,8 +115,14 @@ func extraPasswordFlagsHandlingFuncImpl(c *PasswordCommand, _ *base.FlagSets, op
 				return false
 			}
 			*opts = append(*opts, accounts.WithPasswordAccountPassword(strings.TrimSpace(value)))
+
 		default:
-			*opts = append(*opts, accounts.WithPasswordAccountPassword(c.flagPassword))
+			password, err := parseutil.ParsePath(c.flagPassword)
+			if err != nil && err.Error() != parseutil.ErrNotAUrl.Error() {
+				c.UI.Error("Error parsing password flag: " + err.Error())
```
Comment (irenarindos) [inline internal/cmd/commands/accountscmd/password_funcs.go:122]:
```suggestion
				c.UI.Error(fmt.Sprintf("Error parsing password flag: %s", err.Error()))
```

File: internal/cmd/commands/authenticate/password.go [hunk 5]
```diff
@@ -110,6 +112,14 @@ func (c *PasswordCommand) Run(args []string) int {
 			return base.CommandUserError
 		}
 		c.flagPassword = strings.TrimSpace(value)
+
+	default:
+		password, err := parseutil.ParsePath(c.flagPassword)
+		if err != nil && err.Error() != parseutil.ErrNotAUrl.Error() {
+			c.UI.Error("Error parsing password flag: " + err.Error())
```
Comment (irenarindos) [inline internal/cmd/commands/authenticate/password.go:119]:
```suggestion
			c.UI.Error(fmt.Sprintf("Error parsing password flag: %s", err.Error()))
```

**Review by irenarindos — CHANGES_REQUESTED** [review]

File: internal/cmd/commands/accountscmd/funcs.go (hunk 1 above)
Comment (louisruch) [reply internal/cmd/commands/accountscmd/funcs.go:146]: Or `fmt.Sprintf("Error parsing password flag: %w", err`

**Review by louisruch — APPROVED** [review]
Agree with the `fmt` comment but otherwise LGTM

File: internal/cmd/commands/accountscmd/funcs.go (hunk 1 above)
Comment (jefferai) [reply internal/cmd/commands/accountscmd/funcs.go:146]: You can't use %w with sprintf; going with `c.UI.Error(fmt.Sprintf("Error parsing password flag: %v", err))`

Comment (louisruch) [reply internal/cmd/commands/accountscmd/funcs.go:146]: 👍

**Review by irenarindos — APPROVED** [review]

**Review by johanbrandhorst — APPROVED** [review]

**Pull request merged**